modules/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js

```diff
@@ -536,19 +594,24 @@ return view.extend({
 			uci.unset('dhcp', section, 'dns');
 		};
 
-		so = ss.option(form.Value, 'mac', _('MAC address'));
-		so.datatype = 'list(macaddr)';
+		var macwildcard = function(multicast) {
+			var m = this.value.match(/^([a-fA-F0-9]{2}|\*):([a-fA-F0-9]{2}:|\*:){4}(?:[a-fA-F0-9]{2}|\*)$/);
+			return this.assert(m != null && !(+m[1] & 1) == !multicast,
+				multicast ? _('valid multicast MAC address with wildcard (*)') : _('valid MAC address with wildcard (*)'));
+		}
+
+		so = ss.option(form.Value, 'mac',
+			_('MAC address(es)'),
+			_('The hardware address(es) of this entry/host, separated by spaces.'));
+		so.datatype = 'list(macwildcard)';
 		so.rmempty  = true;
 		so.cfgvalue = function(section) {
 			var macs = L.toArray(uci.get('dhcp', section, 'mac')),
 			    result = [];
 
 			for (var i = 0, mac; (mac = macs[i]) != null; i++)
-				if (/^([0-9a-fA-F]{1,2}):([0-9a-fA-F]{1,2}):([0-9a-fA-F]{1,2}):([0-9a-fA-F]{1,2}):([0-9a-fA-F]{1,2}):([0-9a-fA-F]{1,2})$/.test(mac))
-					result.push('%02X:%02X:%02X:%02X:%02X:%02X'.format(
-						parseInt(RegExp.$1, 16), parseInt(RegExp.$2, 16),
-						parseInt(RegExp.$3, 16), parseInt(RegExp.$4, 16),
-						parseInt(RegExp.$5, 16), parseInt(RegExp.$6, 16)));
+				if (/^([0-9a-fA-F]{1,2}|\*):([0-9a-fA-F]{1,2}|\*):([0-9a-fA-F]{1,2}|\*):([0-9a-fA-F]{1,2}|\*):([0-9a-fA-F]{1,2}|\*):([0-9a-fA-F]{1,2}|\*)$/.test(mac))
+					result.push('%s:%s:%s:%s:%s:%s'.format(RegExp.$1, RegExp.$2, RegExp.$3, RegExp.$4, RegExp.$5, RegExp.$6));
```

Comment:
The changed code will not normalize MACs anymore, e.g.
`0:1:2:3:4:5` -> `00:01:02:03:04:05`
`a:b:11:FF:00:3` -> `AA:BB:11:FF:00:30`
`0:11:a:*:*:*` -> `00:11:0A:*:*:*`

Maybe consider the following:
```javascript
if (/^([0-9a-fA-F]{1,2}|\*):([0-9a-fA-F]{1,2}|\*):([0-9a-fA-F]{1,2}|\*):([0-9a-fA-F]{1,2}|\*):([0-9a-fA-F]{1,2}|\*):([0-9a-fA-F]{1,2}|\*)$/.test(mac)) {
    var m = [
        parseInt(RegExp.$1, 16), parseInt(RegExp.$2, 16),
        parseInt(RegExp.$3, 16), parseInt(RegExp.$4, 16),
        parseInt(RegExp.$5, 16), parseInt(RegExp.$6, 16)
    ];

    result.push(m.map(function(n) { return isNaN(n) ? '*' : '%02X'.format(n) }).join(':'));
}
```

Reply:
This normalization functions properly.